app.js

```diff
@@ -219,28 +222,58 @@ app.post('/interactions', verifyKeyMiddleware(process.env.PUBLIC_KEY), async (re
-            }
-            if (lbargs[lbargs.length - 1] === 'f') {
-                args.push(0);
-            } else if (lbargs[lbargs.length - 1] === 'l') {
-                args.push(1000-args[3]);
-            } else if (lbargs[lbargs.length - 1] === 'p') {
-                data.values[0].split(';').forEach((e) => {
-                    args.push(e);
-                });
-            }
+            if (lbargs[1] === 'totd') track_info.endTimestamp = Number(convertNumberToBase(lbargs[2], 64, 10));
+            let body;
+            if (lbargs[lbargs.length - 1] ==='i') {
+                let leaderboard = await trackmaniaFacade.getLeaderboardInfo(track_info, args[3], true, args[4]);
+                const userId = message.interaction.user.id;
+                const userName = message.interaction.user.global_name;
+                if (accountWatchers[userId] !== undefined) {
+                    const mapType = lbargs[lbargs.length-3];
+                    const watchedRecords = await trackmaniaFacade.getWatchedAccounts(accountWatchers[userId], revertUID(lbargs[lbargs.length-2], mapType));
+                    console.log(watchedRecords);
+                    const watchingLB = await trackmaniaFacade.generateLeaderboardField({
+                        name: `${userName}'s Watched Players`,
+                        records: watchedRecords,
+                    });
+                    leaderboard.fields.push(watchingLB.field);
+                    leaderboard.watchedAccounts = watchingLB.players;
+                }
+                body = trackmania.embedLeaderboardInfo(leaderboard);
+            } else {
+                console.log(lbargs);
+                if (lbargs[lbargs.length - 1] === 'f') {
+                    args.push(0);
+                } else if (lbargs[lbargs.length - 1] === 'l') {
+                    args.push(1000-args[3]);
+                } else if (lbargs[lbargs.length - 1] === 'p') {
+                    data.values[0].split(';').forEach(e => args.push(e));
+                }
 
-            const lb_info = await trackmaniaFacade.getLeaderboardInfo(track_info, args[3], true, args[4]);
+                console.log(args);
 
-            log.info(args);
+                const { records, buttons, pageSelecter } = await trackmaniaFacade.updateLeaderboard({
+                    groupUid: track_info.groupUid,
+                    mapUid: track_info.mapUid,
+                }, Number(args[3]), true, Number(args[4]));
 
+                let embeds = message.embeds;
+                let components = message.components;
 
+                embeds[0].fields[0] = records.field;
+                components[0].components = buttons;
+                components[1].components[0].options = records.players;
+                components[3] = pageSelecter;
+
+                body = {
+                    embeds: embeds,
+                    components: components,
+                };
+            }
```

Comment:
Could we move this logic to some function that builds the embed for Discord? Similar to how the 'i' branch above feeds a model (all of our API info) into the `trackmania.embedLeaderboardInfo` which constructs our "view". This will help keep `app.js` small